bigbluebutton-client/src/org/bigbluebutton/modules/whiteboard/views/WhiteboardCanvas.as

```diff
@@ -97,10 +103,16 @@ package org.bigbluebutton.modules.whiteboard.views {
 			wbModel.addEventListener(WhiteboardAccessEvent.MODIFIED_WHITEBOARD_ACCESS, onModifiedAccess);
 			wbModel.addEventListener(WhiteboardCursorEvent.RECEIVED_CURSOR_POSITION, onReceivedCursorPosition);
 			
+			whiteboardToolbar.addEventListener(WhiteboardAccessEvent.MODIFY_WHITEBOARD_ACCESS, onMultiUserBtn);
 			whiteboardToolbar.addEventListener(WhiteboardButtonEvent.ENABLE_WHITEBOARD, onEnableWhiteboardEvent);
 			whiteboardToolbar.addEventListener(WhiteboardButtonEvent.DISABLE_WHITEBOARD, onDisableWhiteboardEvent);
-			whiteboardToolbar.addEventListener(WhiteboardAccessEvent.MODIFY_WHITEBOARD_ACCESS, onMultiUserBtn);
-
+			
+			var stpl:Listener = new Listener();
+			stpl.type = SwitchedPresenterEvent.SWITCHED_PRESENTER;

```

Comment:
The presenter switching doesn't work like this. The WhiteboardCanvas is notified about who the pod presenter is from the PresentationPod. The even you're listening for here is the global presenter which is a completely different concept.